cdap-app-fabric/src/main/java/io/cdap/cdap/internal/operation/LongRunningOperation.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.operation;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.cdap.cdap.proto.operation.OperationError;
+import io.cdap.cdap.proto.operation.OperationMeta;
+import java.util.function.Consumer;
+
+/**
+ * LongRunningOperation represents a long-running asynchronous operation.
+ */
+public interface LongRunningOperation {
```

Comment:
Not sure if `runner` prefix is a good fit here. A implementation of a `LongRunningOperation` represents a operation type and its instance represents a specific operation run. 
`LongRunningOperation` is analogous to `RunnableTask` here. Which has similar class hierarchy and semantics.
IMHO `runner` feels more like a class which actually runs a operation rather than which contains the operation logic itself.